src/Microsoft.AspNet.Razor/CodeGenerators/CSharpTagHelperCodeRenderer.cs

```diff
@@ -688,6 +712,21 @@ private void RenderBufferedAttributeValueAccessor(CSharpCodeWriter writer)
             }
         }
 
+        private static bool ContainsChildContent(IList<Chunk> children)
+        {
+            foreach (var child in children)
+            {
+                if (!(child is ParentChunk) ||
+                    child is TagHelperChunk ||
+                    ContainsChildContent(((ParentChunk)child).Children))
```

Comment:
Since `ParentChunk` is a `class`, also more readable (and better in general) to use `as` and a temporary variable than to do the cast twice.